File: day-1/python/solution.py
with open('../inputs/input.txt', 'r') as f:
    loads = [load.split("\n") for load in f.read().strip().split("\n\n")]
    calories_per_elf = [sum([int(cals) for cals in load]) for load in loads]
    biggest_caloric_load = sorted(calories_per_elf)[-1]
    top_three_caloric_loads = sum(sorted(calories_per_elf)[-3:])
    print("The largest load was {:,} calories.".format(biggest_caloric_load))
    print("The top 3 largest loads were: {:,}.".format(top_three_caloric_loads))

File: day-1/ruby/solution.rb
lines = File.open('../inputs/test.txt').read.split("\n\n")
calories_per_elf = lines.map { |l| l.split("\n").map(&:to_i).sum}.sort
puts "The highest calorie load was #{calories_per_elf.last}."
puts "The top 3 highest calorie loads were: %d, %d, and %d." % calories_per_elf.last(3)


